src/rgw/rgw_cache.h

```diff
@@ -182,8 +182,8 @@ class RGWCache  : public T
 
   string normal_name(rgw_pool& pool, const std::string& oid) {
     std::string buf;
-    buf.reserve(pool.name.size() + oid.size() + 2);
-    buf.append(pool.name).append("+").append(oid);
+    buf.reserve(pool.name.size() + pool.ns.size() + oid.size() + 2);
+    buf.append(pool.name).append("+").append(pool.ns).append("+").append(oid);
```

Comment:
just a note that, although this adds an extra `"+"` to the string without accounting for it in the math for `buf.reserve()`, the result is still correct because `buf.reserve()` already accounts for the null terminator